### ISSUE TITLE
Is the datastore built using the validation set?

### ISSUE THREAD

**Issue by SirRob1997** [zhengxxn/adaptive-knn-mt#2]
While looking through the `save_datastore.py`, I found [this line](https://github.com/zhengxxn/adaptive-knn-mt/blob/5ee6581ec28d0aaff4b1013140555a9ef4d182c0/save_datastore.py#L96) which reads rather odd together with `dataset = task.dataset(subset)`. It seems like the datastore is built using the validation set or am I misinterpreting something here? It should be built using the training data, right?

Since the implementation is based on `validate.py` maybe this was falsely not adapted when creating the script. I believe it should be using `args.train_subset` instead.

**Comment by zhengxxn** [comment]
we set "train" for the "--valid-subset" parameter, and you can find it in the "Create Datastore" script. So it actually uses the train set to build the datastore.

**Comment by SirRob1997** [comment]
Ah I see, missed that! Thanks for clarifying.

**Comment by SirRob1997** [comment]
Probably on a related note: Why do you pass `--train-subset valid` in the training script?

**Comment by zhengxxn** [comment]
Since the `training set` for training the `meta-k network` in our paper should not include the sentences that already used for building the datastore ( if this happened, then the 1-NN retrieval result for the `training set` is always the best choice).

A probable practice is to split the training set into two parts: one for building the datastore, and the other for training the `meta-k` network.  But, this `meta-k network` is very simple and we found that only a few training sentences are enough.
So for simplicity, we used the full `train set` to build datastore and directly used the `valid set` to train the `meta-k` network.
